content/docs/guides/components/istio.md

```diff
@@ -13,63 +13,24 @@ toc = true
-    ```
-
-### 2. Configure sidecar injection
-Before installing Istio components, we will change some configuration.
-We will use isio's automatic sidecar injection. However, we want to inject only to those services with annotation.
-According to this [table](https://github.com/istio/istio/issues/6476#issuecomment-399219937), we will
-label the namespace as `enabled`, and set policy in `istio-sidecar-injector` (a configmap) as `disabled`.
-
-- label the namespace
-   ```
-   kubectl label namespace kubeflow istio-injection=enabled
-   ```
-- Set the policy. This can be done by editing `install/kubernetes/istio-demo.yaml` directly.
-
-
-### 3. Configure egress traffic
-Also, we need to allow egress traffic, e.g. read models from GCS. This is control by the `-i` flag of the args of 
-`istio-init` container in the same configmap `istio-sidecar-injector`. We can also edit it directly.
-To determine the IP values, follow the instructions in the Istio
-[doc](https://istio.io/docs/tasks/traffic-management/egress/#calling-external-services-directly). For example, for GKE you can do
+We assume Kubeflow is already deployed in the `kubeflow` namespace.
 
 ```
-gcloud container clusters describe XXXXXXX --zone=XXXXXX | grep -e clusterIpv4Cidr -e servicesIpv4Cidr
-clusterIpv4Cidr: 10.4.0.0/14
-servicesIpv4Cidr: 10.7.240.0/20
-```
-
-- Set the outbound IP ranges (direcly in the configmap). The value should be like
-   `"10.32.0.0/14\,10.35.240.0/20"`.
+kubectl apply -f https://raw.githubusercontent.com/kubeflow/kubeflow/master/dependencies/istio/install/crds.yaml
+kubectl apply -f https://raw.githubusercontent.com/kubeflow/kubeflow/master/dependencies/istio/install/istio-noauth.yaml
+kubectl apply -f https://raw.githubusercontent.com/kubeflow/kubeflow/master/dependencies/istio/kf-istio-resources.yaml
 
-### 4. Configure istio-ingressgateway
-We recommend using a regular ingress instead of Istio's ingress gateway as the entry to cluster on GCP,
-so this service should not be of type `LoadBalancer`.
-
-- Edit `install/kubernetes/istio-demo.yaml` and change the service type of `istio-ingressgateway` from
-  `LoadBalancer` to `NodePort`.
-
-### 4. Install istio (without mTLS)
-    ```
-    kubectl apply -f install/kubernetes/istio-demo.yaml
-    ```
+kubectl label namespace kubeflow istio-injection=enabled
+```
 
-### (optional) 5. Deploy the Gateway
+The first command installs Istio's CRDs.
 
-This is for rolling out model and doing traffic split. See more detail below.
+The second command installs Istio's core components (without mTLS), with some customization:
+1. sidecar injection configmap policy is changed from `enabled` to `disabled`
+1. istio-ingressgateway is of type `NodePort` instead of `LoadBalancer`
 
-TODO(https://github.com/kubeflow/kubeflow/issues/1309): update this
+The third command deploys some resources for Kubeflow.
+The fourth command label the kubeflow namespace for sidecar injector.
```

Comment:
Maybe better to say, "enables the sidecar for pods in the kubeflow namespace".

Thinking about this, do you want all pods in that namespace to have the sidecar. Would it be better to make a kubeflow-serving namespace?

Reply:
No it's not injecting all pods in kubeflow namespace.
It only injects when your deployment has the proper annotation.
I think this is the behavior we want.